src/solc_0.8/asset/ERC1155ERC721.sol

```diff
@@ -54,34 +53,47 @@ contract ERC1155ERC721 is WithSuperOperators, IERC1155, IERC721 {
 
     mapping(address => address) private _creatorship; // creatorship transfer
 
+    // @review Bouncer
     mapping(address => bool) private _bouncers; // the contracts allowed to mint
-    mapping(address => bool) private _metaTransactionContracts; // native meta-transaction support
-
+    mapping(address => bool) private _metaTransactionProcessors; // meta-transaction support
+    // @review Bouncer ... Deprecate?
     address private _bouncerAdmin;
 
     bool internal _init;
 
+    // Enables version lookup programatically
+    uint256 public constant _version = 2;
```

Comment:
you mentioned this in one of our call. 
What the benefit over tracking version off-chain ?
We store all our deployment in git and can technically find matching addresses
we could also make sure that all previous implementation deployment are kept in separate file in the deployments folder
hardhat-deploy also keep track of history when using it for upgrade (something I did not a chance yet to explore for openzeppelin support)

Reply:
Thinking about this some more, I think the only benefit would be if we foresee a need to query the version from another contract. In the context of the sandbox Asset, I don't see that we would need to ever support more than 1 version of the contract at a time, so this benefit will probably never be realized. 

I think having the version appended to the name, (ie: AssetV2) in addition to the mechanisms you mentioned above should be completely adequate (and cheaper).